visualization/app/codeCharta/state/settings.service.ts

```diff
@@ -13,41 +13,39 @@ export interface SettingsServiceSubscriber {
 
 export class SettingsService implements FileStateServiceSubscriber {
 	private static SETTINGS_CHANGED_EVENT = "settings-changed"
+	private static DEBOUNCE_TIME = 400
 
 	private settings: Settings
-	private readonly debounceBroadcast: (update: RecursivePartial<Settings>) => void
+	private update: RecursivePartial<Settings> = {}
+	private readonly debounceBroadcast: () => void
 
 	constructor(private $rootScope: IRootScopeService, private $timeout: ITimeoutService, private loadingGifService: LoadingGifService) {
 		this.settings = this.getDefaultSettings()
-		this.debounceBroadcast = _.debounce(
-			(update: RecursivePartial<Settings>) =>
-				this.$rootScope.$broadcast(SettingsService.SETTINGS_CHANGED_EVENT, { settings: this.settings, update: update }),
-			400
-		)
+		this.debounceBroadcast = _.debounce(() => {
```

Comment:
Do we wanna separate the two broadcast lines into an extra function?

Reply:
Moved the broadcast and reset of `this.update` into a function `notifySubscribers`. I couldn't find a better fitting name and I think `notifySubscribersAndResetUpdateState` is too much